src/main/java/net/neoforged/neoforge/client/entity/animation/json/JsonAnimationLoader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation.json;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.mojang.logging.LogUtils;
+import java.util.HashMap;
+import java.util.Map;
+import net.minecraft.client.animation.AnimationDefinition;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+/**
+ * A loader for entity animations written in JSON. You can also get parsed animations from this class, but the
+ * recommended way to use JSON entity animations is with {@link JsonAnimator}.
+ *
+ * @see JsonAnimator
+ */
+public final class JsonAnimationLoader extends SimpleJsonResourceReloadListener {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public static final JsonAnimationLoader INSTANCE = new JsonAnimationLoader();
+
+    private final Map<ResourceLocation, AnimationDefinition> animations = new HashMap<>();
+
+    private JsonAnimationLoader() {
+        super(new Gson(), "animations/entity");
+    }
+
+    /**
+     * Get the animation with the specified ID.
+     * 
+     * @param animationId The ID of the animation.
+     * @return The animation, or null if not found.
+     */
+    @Nullable
+    public AnimationDefinition getAnimation(ResourceLocation animationId) {
+        return animations.get(animationId);
+    }
```

Comment:
Do we want to return the raw animation definition, or should we store this in some sort of holder or supplier so that any resource reloads can be properly handled. I imagine most people will simply use a static instance to store this definition.

Reply:
The primary way to use this API, as per `JsonAnimator` is to only store `ResourceLocation`s. This method is only if you want to perform some kind of introspection on the currently loaded definition.